src/arviz_stats/loo/loo_moment_match.py

```diff
@@ -0,0 +1,792 @@
+"""Compute moment matching for problematic observations in PSIS-LOO-CV."""
+
+import warnings
+from collections import namedtuple
+from copy import deepcopy
+
+import numpy as np
+import xarray as xr
+from arviz_base import rcParams
+from xarray_einstats.stats import logsumexp
+
+from arviz_stats.loo.helper_loo import (
+    _get_log_likelihood_i,
+    _get_r_eff,
+    _prepare_loo_inputs,
+    _recalculate_weights_k,
+    _shift,
+    _shift_and_cov,
+    _shift_and_scale,
+    _update_loo_data_i,
+)
+from arviz_stats.sampling_diagnostics import ess
+from arviz_stats.utils import ELPDData
+
+SplitMomentMatch = namedtuple("SplitMomentMatch", ["lwi", "lwfi", "log_liki", "reff"])
+
+
+def loo_moment_match(
+    data,
+    loo_orig,
+    upars,
+    log_prob_upars_fn,
+    log_lik_i_upars_fn,
+    max_iters=30,
+    k_threshold=None,
+    split=True,
+    cov=False,
+    pointwise=None,
+    var_name=None,
```

Comment:
not sure how strict we are about this here (we are more consistent in arviz-plots), but usually var_name/var_names is listed among the first few arguments